srcpkgs/PWAsForFirefox/template

```diff
@@ -0,0 +1,36 @@
+# Template file for 'PWAsForFirefox'
+pkgname=PWAsForFirefox
+version=2.7.3
+revision=1
+archs="~*-musl"
+build_style=cargo
+hostmakedepends="pkg-config"
+makedepends="openssl-devel"
+depends="firefox"
+short_desc="Native part of the firefox extension PWAsForFirefox"
+maintainer="Feleuxens <me@feleuxens.de>"
+license="MPL-2.0"
+homepage="https://github.com/filips123/PWAsForFirefox"
+distfiles="https://github.com/filips123/PWAsForFirefox/archive/refs/tags/v${version}.tar.gz"
+checksum=b520cf5caeeca1d23043a032137f7eead7eb88270e5376c5d08b1234bb90376f
+
+post_extract() {
+	# rm everything except native/
+	find . -mindepth 1 -maxdepth 1 -not -name native -exec rm -r '{}' \;
+	# move everything inside native to root
+	mv native/* .
+	# set version because it is set to 0.0.0
+	sed -i "s/0.0.0/${version}/" Cargo.toml
+	sed -i "/firefoxpwa/,/0.0.0/ s/0.0.0/${version}/" Cargo.lock
+	sed -i "s/DISTRIBUTION_VERSION = \"0.0.0\"/DISTRIBUTION_VERSION = \"${version}\"/g" userchrome/profile/chrome/pwa/chrome.jsm
+}
+
+do_install() {
+	install -D "target/${RUST_TARGET}/release/firefoxpwa" "${DESTDIR}/usr/bin/firefoxpwa"
```

Comment:
All this should be vinstall, see https://github.com/void-linux/void-packages/blob/master/Manual.md#global-functions